File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        temp=list1
        temp1=list2
        ans=ListNode(0)
        ptr=ans
        while (temp and temp1):
            if temp.val > temp1.val:
                ptr.next=ListNode(temp1.val)
                ptr=ptr.next
                temp1=temp1.next
            else:
                ptr.next=ListNode(temp.val)
                ptr=ptr.next
                temp=temp.next
        while temp:
            ptr.next=ListNode(temp.val)
            ptr=ptr.next
            temp=temp.next
        while temp1:
            ptr.next=ListNode(temp1.val)
            ptr=ptr.next
            temp1=temp1.next
            
        return ans.next
            

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode *l3 = new ListNode(0);
        ListNode* head = l3;
        while(list1 && list2){
            if(list1->val < list2->val)
            {
                l3->next= new ListNode(list1->val);
                list1=list1->next;
                l3=l3->next;
            }
            else{
                l3->next= new ListNode(list2->val);
                list2=list2->next;
                l3=l3->next;
            }
           
        } 
        while(list1){
                 l3->next= new ListNode(list1->val);
                list1=list1->next;
                l3=l3->next;
            }
              while(list2){
                 l3->next= new ListNode(list2->val);
                list2=list2->next;
                l3=l3->next;
            }
            return head->next;
    }
};